### PR TITLE
Setup container test for nodejs

### PR DESCRIPTION
**Pull request by neomatrix369**
Docker container setup with 
- node
- yarn
- prettier

Based on the Docker/stretch container for nodejs, except we are not using root-less user `node` but as `root`

We haven't applied any flag to ignore failed tests, so that the test-runner does not exit at the first failing test and coverage plugin gets the full picture.

### CODE REVIEW

File: container/images/nodejs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
+  && curl -fsSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
+  && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
+  && mkdir -p /opt \
+  && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/ \
+  && ln -s /opt/yarn-v$YARN_VERSION/bin/yarn /usr/local/bin/yarn \
+  && ln -s /opt/yarn-v$YARN_VERSION/bin/yarnpkg /usr/local/bin/yarnpkg \
+  && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz
+
+RUN npm install -g prettier
```
Comment (julianghionoiu) [inline container/images/nodejs/Dockerfile]: This dependency is not needed if we use the `prettier` from `node_modules`.

File: container/images/nodejs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  && case "${dpkgArch##*-}" in \
+    amd64) ARCH='x64';; \
+    ppc64el) ARCH='ppc64le';; \
+    s390x) ARCH='s390x';; \
+    arm64) ARCH='arm64';; \
+    armhf) ARCH='armv7l';; \
+    i386) ARCH='x86';; \
+    *) echo "unsupported architecture"; exit 1 ;; \
+  esac \
+  && curl -fsSLO --compressed "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-$ARCH.tar.xz" \
```
Comment (julianghionoiu) [inline container/images/nodejs/Dockerfile]: Installing Node seems so complicated. Is there a node version available in `apt`?

Comment (neomatrix369) [reply container/images/nodejs/Dockerfile]: Same response as in my previous PR response, I can use apt-get although we need certain versions and what if we can;t get them via apt-get, how dependent are we on the node/npm versions?

Comment (julianghionoiu) [reply container/images/nodejs/Dockerfile]: With Python, we can use any version of Python 2.7+.  For Python 3 we need to make some changes to the runner to make it compatible.

Comment (julianghionoiu) [reply container/images/nodejs/Dockerfile]: You can use any nodejs version provided that it is supported by Codeship: https://documentation.codeship.com/basic/languages-frameworks/nodejs/

Comment (neomatrix369) [reply container/images/nodejs/Dockerfile]: Just had a lookout for node or npm or nvm via `apt-get` no success

Comment (neomatrix369) [reply container/images/nodejs/Dockerfile]:
```
root@950dc6d77b96:/srv# nvm --version
0.33.11
root@950dc6d77b96:/srv# npm --version
3.6.0
root@950dc6d77b96:/srv# node --version
v5.6.0
```

**Pull request merged**